R/summary_forest_plot.R

```diff
@@ -0,0 +1,637 @@
+      lnticks <- log(vticks)
+      lblticks <- c("1/1024", "1/256", "1/64", "1/16", "1/4", "1", "4", "16", "64", "256", "1024")
+      axis(1, at = lnticks, labels = lblticks, cex.axis = 0.6, padj = -1.0, tck = -0.05)
+    } else { #continuous data scale
+      axis(1, at = NULL, labels = TRUE, cex.axis = 0.6, padj = -1.0, tck = -0.05)
+    }
+  }
+}
+
+#' Function to draw graphs along diagonal - for NMA SPF Matrix.
+#'
+#' @param ntx Number of treatments
+#' @param rkgram Vector containing rankogram data
+#' @param cumu True if to draw cumulative rankogram. Defaults to False
+rankogram <- function(ntx, rkgram, cumu = FALSE) {
+  ori.ntx <- length(rkgram) / ntx
+  xseq <- seq(0, 1, length.out = (2 * ntx + 1))
+  rankmat <- array(rkgram, c(ori.ntx, ntx))
+  
+  if (cumu) {
+    rank.cumprob <- apply(rankmat, 2, cumsum)   #2:indicates column to all apply to
+  }
+  
+  for (i in 1:ntx) {
+    if (cumu) {
+      par(fig = c(xseq[2 * i - 1], xseq[2 * i + 1], (1 - xseq[2 * i + 1]), (1 - xseq[2 * i])), new = TRUE, mar = c(1.2, 1.5, 0, 0.6))
+    } else {
+      par(fig = c(xseq[2 * i - 1], xseq[2 * i + 1], (1 - xseq[2 * i + 1]), (1.01 - xseq[2 * i])), new = TRUE, mar = c(1.2, 1.5, 0, 0.6))
+    }
+    plot(1:ori.ntx, seq(-1, 1, len = ori.ntx), type = "n",  ylab = "", xlab = "", ylim = c(0, 1), axes = FALSE)
+    
+    if (!cumu) {
+      lines(1:ori.ntx, rankmat[, i], lwd = 1.8)
+    } else {
+      lines(c(1, c(1:c(ori.ntx - 1)) + 0.5, ori.ntx), rank.cumprob[c(1, 1:c(ori.ntx - 1), ori.ntx), i])
+    }
+    
+    xticks <- c(1:ori.ntx)
+    xlblticks <- c(1, rep("", (ori.ntx - 2)), ori.ntx)
+    axis(1, at = xticks, labels = xlblticks, lwd = 0.8, cex.axis = 0.7, tck = -0.05, padj = -2, col = 'grey70', col.axis = 'grey70') #bottom axis
+    yticks <- seq(0.0, 1.0, by = 0.5)
+    axis(2, at = yticks, labels = yticks, lwd = 0.8, cex.axis = 0.7, las = 2, tck = -0.06, hadj = 0.2, line = -0.2, col = 'grey70', col.axis = 'grey70')  #left axis
+  }
+}
+
+#' Function to create a vector indexing shading in mtcMultiplot.
+#'
+#' @param ntx Number of treatments
+#'
+#' @return the calculated shading vector
+shading.vec <- function(ntx) {
+  ordvec <- seq(1, ntx * ntx)
+  shgvec <- rep(0, ntx * ntx)
+  
+  #Odd Number interventions
+  shgvec[(ntx %% 2 != 0) & ordvec %% 2 != 0] <- 1
+  
+  #Even Number interventions
+  shgvec[(ntx %% 2 == 0) & ((ceiling(ordvec / ntx)) %% 2 == 0) & (ordvec %% 2 == 0)] <- 1
+  shgvec[(ntx %% 2 == 0) & ((ceiling(ordvec / ntx)) %% 2 != 0) & (ordvec %% 2 != 0)] <- 1
+  
+  return(shgvec)
+}
+
+
+#' Draw the elements of the summary forest matrix plot.
+#'
+#' @param stytitle Title of plot
+#' @param ntx Number of treatments
+#' @param lstx Vector of treatment names
+#' @param mtc Meta-analysis data for direct and indirect evidence
+#' @param ma Meta-analysis data for direct evidence only
+#' @param bpredd TRUE if confidence interval to be plotted as error bars
```

Comment:
I think this should be 'predictive interval' (based on name)
Same on line 579

Reply:
Fixed